crates/webidl/src/util.rs

```diff
@@ -707,3 +710,26 @@ pub fn public() -> syn::Visibility {
         pub_token: Default::default(),
     })
 }
+
+/// When generating our web_sys APIs we default to setting slice references that
+/// get passed to JS as mutable in case they get mutated in JS.
+///
+/// In certain cases we know for sure that the slice will not get mutated - for
+/// example when working with the WebGlRenderingContext APIs.
+///
+/// Here we implement a whitelist for those cases. This whitelist is currently
+/// maintained by hand.
+fn maybe_adjust<'a> (idl_type: IdlType<'a>, id: &'a OperationId) -> IdlType<'a> {
+    match id {
+        // TODO: `match op` and return an adjusted idl_type if necessary
+        OperationId::Operation(Some(op)) => {
+            match *op {
+//                "vertexAttrib1fv" => {
```

Comment:
I think that here the input type needs to be a `Float32Array` because otherwise it's mapping *all* arguments to a `Float32Array`, and I think that's causing problems?

Reply:
Ohhhhh - thanks I'll try that next time I sit down on this!